### PR TITLE
GS-101: Implement auth state changed

### PR DESCRIPTION
**Pull request by gochoa1088**
This PR closes out https://spacem8s.atlassian.net/browse/GS-101

### CODE REVIEW

File: components/ColorFilter/index.js [hunk 1]
```diff
@@ -8,8 +6,10 @@ const ColorFilter = () => {
     <>
       <p className={styles.subheading}>Filter by Color: </p>
       <div className={styles.container}>
-        {colors.map((color) => (
-          <div className={styles.color}>{color}</div>
+        {colors.map((color, index) => (
```
Comment (gochoa1088) [inline components/ColorFilter/index.js:9]: change how the key works

File: components/CreateAccountForm/index.js [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
+import { useRouter } from "next/router";
 
-import { auth } from "../../firebase.js";
+import { useAuth } from "../../context/AuthContext.js";
 
 import styles from "./styles.module.css";
 
 const CreateAccountForm = () => {
+  const { user, signup } = useAuth();
+  console.log(user);
```
Comment (gochoa1088) [inline components/CreateAccountForm/index.js:10]: remove comment

File: components/CreateAccountForm/index.js [hunk 3]
```diff
@@ -1,22 +1,28 @@
...
+  const handleCreateAccount = async (e) => {
+    e.preventDefault();
+
+    try {
+      await signup(email, password);
+      router.push("/today");
+    } catch (err) {
+      console.log(err);
+    }
+    console.log(email, password);
```
Comment (gochoa1088) [inline components/CreateAccountForm/index.js:25]: remove comment

File: context/AuthContext.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+import { auth } from "../firebase";
+
+const AuthContext = createContext();
+
+export const useAuth = () => useContext(AuthContext);
+
+export const AuthContextProvider = ({ children }) => {
+  const [user, setUser] = useState(null);
+  const [loading, setLoading] = useState(true);
+  console.log("inside context provider", user);
```
Comment (gochoa1088) [inline context/AuthContext.js:17]: remove console.log

**Review by vincentochoa3 — APPROVED** [review]
Looks good and thanks for cleaning up/catching those errors

**Pull request merged**